tools/residency_contract/schemas.py

```
@@ -0,0 +1,1125 @@
+        raw_predicate = _mapping(conditional.get(key), f"schema conditional {key}")
+        predicate = _predicate(raw_predicate)
+        if key == "unless":
+            predicate = {"not": predicate}
+        translated = {"if": predicate, "then": _consequence(conditional)}
+        if "contexts" in raw_predicate:
+            translated["x-residency-contexts"] = raw_predicate["contexts"]
+        result.append(translated)
+    return result
+
+
+def _required_schema_keywords(value: Any) -> list[str]:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)